packages/@aws-cdk/aws-servicecatalogappregistry/lib/target-application.ts

```diff
@@ -90,7 +91,7 @@ class CreateTargetApplication extends TargetApplication {
     super();
   }
   public bind(scope: Construct): BindTargetApplicationResult {
-    const stackId = this.applicationOptions.stackId ?? 'ApplicationAssociatorStack';
+    const stackId = this.applicationOptions.stackId ?? `CreateTargetApplication${this.applicationOptions.applicationName}`;
```

Comment:
After thinking about it a little, I think this is the wrong way to go about it. A `Stack` has two identifiers:

* `id`, used as an ID in the construct tree (needs to be unique within the scope in the construct tree)
* `stackName`, the name of the CloudFormation Stack being deployed (needs to be unique within the account+region it is deployed to, defaults to `id` if not supplied).

I think what is happening here is that you are trying to change `stackName` by changing `id`, is that right?

If that is the case, doesn't it make more sense to leave `id` where it is, and explicitly set `stackName` to the value you're trying to achieve?

-----

Also, `CreateXxx` is not a great name for a resource (Create is a Verb, but a resource is a Noun). Maybe make it something `Application-MyGreatApplication-Stack` ?

Reply:
Yes you are correct, we are trying to change `stackName` by changing the `id`. Your proposed approach of setting the `stackName` makes more sense and I have updated the commit with this change instead.

I also updated the stackName format as per your suggestion.

----

For your first concern about "renaming" the stack, it is true that if the user wants to continue to rely on the default stack name and keep the same application name, the user will see a conflict because the new stack with the updated name will attempt to create an app that already exists. 

I think a workaround for the user would be to specify the previous default stack name `ApplicationAssociatorStack ` as a `stackName` option property to keep using the old stack name.

Reply:
> I think a workaround for the user would be to specify the previous default stack name ApplicationAssociatorStack  as a stackName option property to keep using the old stack name.

👍 Sounds good. 

The next question is: how will users that get affected by this know that they have to do that?

Here is my suggestion: they will probably read the change log and find this PR. Please edit the description of this PR and add a section that says "if you run into the following error (...error message here...) make sure to set stackName to the name of your stack. For example:

```ts
new ApplicationAssociator(..., {
  stackName: 'whatever-theold-<your application name here>-stackname-was'
```
"

Something like that.

Reply:
That makes a lot of sense, I've added this section in the PR description.